### ISSUE TITLE
When unticking the "Show downloaded" box it sometimes stays disabled.

### ISSUE THREAD

**Issue by Lux-Hue** [axcore/tartube#410]
When I untick the "Show downloaded" box in the toolbar at the bottom of the Videos tab, that entire row of checkboxes become disabled or "greyed out" if there are no undownloaded videos in the selected folder. If I select a different channel or folder, it stays disabled, unless the newly selected channel or folder has one or more undownloaded videos. Then I can tick/untick the checkboxes again. Restarting tartube also enables that row of checkboxes, even though it remembers the "Show downloaded" box was unticked. However, clicking on a folder or channel immediately disables the checkboxes again.

Also, if I untick both "Show downloaded" and "Show undownloaded", there is no way to enable that row of checkboxes again, until I restart tartube.

### What operating system are you using?
Running on Linux Mint 20.3, with unofficial 5.17.3 kernel.

### What version of Tartube are you using?
Version 2.4.0, installed via the .deb package.

### What happens when you run Tartube from a terminal window?
This is GUI specific issue, I don't believe the terminal has anything useful to say.

(Edit) I've run it through the terminal anyway, and this is what it says:
```
/usr/lib/python3/dist-packages/gi/overrides/Gio.py:44: Warning: unable to set property 'strikethrough-set' of type 'gboolean' from value of type 'gchararray'
  return Gio.Application.run(self, *args, **kwargs)
/usr/lib/python3/dist-packages/gi/overrides/Gio.py:44: Warning: unable to set property 'underline-set' of type 'gboolean' from value of type 'gchararray'
  return Gio.Application.run(self, *args, **kwargs)
/usr/lib/python3/dist-packages/gi/overrides/Gio.py:44: Warning: unable to set property 'weight-set' of type 'gboolean' from value of type 'gchararray'
  return Gio.Application.run(self, *args, **kwargs)
/usr/lib/python3/dist-packages/gi/overrides/Gio.py:44: Warning: unable to set property 'style-set' of type 'gboolean' from value of type 'gchararray'
  return Gio.Application.run(self, *args, **kwargs)

(tartube:321796): Gdk-CRITICAL **: 18:25:18.667: gdk_window_thaw_toplevel_updates: assertion 'window->update_and_descendants_freeze_count > 0' failed
```

**Comment by ceonelson** [comment]
Just encountered this as well, closed and reopened Tartube, then immediately checked show downloaded and got this in terminal:

  File "C:/Users\Owner\Documents\___Apps\Tartube\msys64\home\user\tartube\tartube/mainwin.py", line 19456, in on_draw_downloaded_checkbutton_changed
    self.video_catalogue_redraw_all(
  File "C:/Users\Owner\Documents\___Apps\Tartube\msys64\home\user\tartube\tartube/mainwin.py", line 9770, in video_catalogue_redraw_all
    return self.system_error(
AttributeError: 'MainWin' object has no attribute 'system_error'


I then tried to search for a video (which wasn't found), so I hit the X to clear the search filter but nothing happened and nothing in terminal.

I just noticed errors tab though and it said "#175 cancel filter request failed sanity check" - kudos for enhanced error messages :)

Close and relaunch and filter box is empty.

Thanks Axcore!

**Comment by axcore** [comment]
Fixed in v2.4.005.
